app/code/Magento/Directory/Block/Data.php

```diff
@@ -152,31 +152,38 @@ public function getRegionCollection()
     }
 
     /**
+     * @param null|string $defValue
+     * @param string $name
+     * @param string $id
+     * @param string $title
      * @return string
      */
-    public function getRegionHtmlSelect()
+    public function getRegionHtmlSelect($value = null, $name = 'region', $state = 'state', $title = 'State/Province')
     {
         \Magento\Framework\Profiler::start('TEST: ' . __METHOD__, ['group' => 'TEST', 'method' => __METHOD__]);
         $cacheKey = 'DIRECTORY_REGION_SELECT_STORE' . $this->_storeManager->getStore()->getId();
+        if ($value === null) {
+            $value = $this->getRegionId();
+        }
         $cache = $this->_configCacheType->load($cacheKey);
         if ($cache) {
-            $options = unserialize($cache);
+            $options = $this->getSerializer()->unserialize($cache);
         } else {
             $options = $this->getRegionCollection()->toOptionArray();
-            $this->_configCacheType->save(serialize($options), $cacheKey);
+            $this->_configCacheType->save($this->getSerializer()->serialize($options), $cacheKey);
         }
         $html = $this->getLayout()->createBlock(
-            'Magento\Framework\View\Element\Html\Select'
+            \Magento\Framework\View\Element\Html\Select::class
         )->setName(
-            'region'
+            $name
         )->setTitle(
-            __('State/Province')
+            __($title)
         )->setId(
-            'state'
+            $state
```

Comment:
Why not call this variable $id if it sets the id? $state is kind of misleading i think. In getCountryHtmlSelect $id = 'country'.

Reply:
I updated this changes but in code review it would be fail due to sort name for variable.